### PR TITLE
add types,commands,migrations for work with postgresql DBMS

### PR DESCRIPTION
**Pull request by kruatech**
Add types and functions for work with postgresql DBMS.

### CODE REVIEW

File: framework/db/QueryBuilder.php [hunk 1]
```diff
@@ -529,6 +529,42 @@ public function createIndex($name, $table, $columns, $unique = false)
...
+     * Creates a SQL command for creating a new postgresql index.
+     * @param string $name the name of the index. The name will be properly quoted by the method.
+     * @param string $table the table that the new index will be created for. The table name will be properly quoted by the method.
+     * @param string|array $columns the column(s) that should be included in the index. If there are multiple columns, please separate them
+     * by commas. The column names will be properly quoted by the method.
+     * @param string $method to set access method can be:btree,hash,gist,gin,spgist
+     * @return $this the command object itself
+     * This function can be used only postgresql.
+     */  
+    public function createPgIndex($name, $table, $columns, $method = 'btree')
```
Comment (klimov-paul) [inline framework/db/QueryBuilder.php]: How this is different from `yii\db\pgsql\QueryBuilder::createIndex()` ?

File: framework/db/QueryBuilder.php [hunk 2]
```diff
@@ -529,6 +529,42 @@ public function createIndex($name, $table, $columns, $unique = false)
...
+     * by commas. The column names will be properly quoted by the method.
+     * @return $this the command object itself
+     * This function can be used only postgresql.
+     */  
+    public function createPgUnique($name, $table, $columns)
+    {
+        return ('ALTER TABLE ')
+            . $this->db->quoteTableName($table). ' ADD CONSTRAINT '
+            . $this->db->quoteTableName($name) . ' UNIQUE '            
+            . ' (' . $this->buildColumns($columns) . ')';
```
Comment (klimov-paul) [inline framework/db/QueryBuilder.php]: Does result of such DDL will be different from `CREATE UNIQUE INDEX `?

File: framework/db/QueryBuilder.php (hunk 1 above)
Comment (diselop) [inline framework/db/QueryBuilder.php]: method access, in postgresql can be btree,hash,gist,gin,spgist.

File: framework/db/Schema.php [hunk 3]
```diff
@@ -57,6 +57,45 @@
     const TYPE_BINARY = 'binary';
     const TYPE_BOOLEAN = 'boolean';
     const TYPE_MONEY = 'money';
+    //Postgresql types
```
Comment (klimov-paul) [inline framework/db/Schema.php]: Such addition unacceptable: `Schema` can provide only general types.

File: framework/db/SchemaBuilderTrait.php [hunk 4]
```diff
@@ -258,4 +258,355 @@ public function money($precision = null, $scale = null)
         }
         return $this->getDb()->getSchema()->createColumnSchemaBuilder(Schema::TYPE_MONEY, $length);
     }
+
```
Comment (klimov-paul) [inline framework/db/SchemaBuilderTrait.php]: Same as above.

**Comment by cebe** [comment]
> Add types and functions for work with postgresql DBMS.

The purpose of the columnschema builder is to create dbms independent schema. If you want to use postgres only, you can just define the column type as string using postgres types directly.

Also, we have subclasses for each DBMS for the querybuilder so dbms specific things should go there.

I'm sorry but we can not take this PR. I am open to discuss parts of this code to be applied in other places but I'll close this PR for now.

**Pull request closed without merging**